opm/common/data/SimulationDataContainer.cpp

```diff
@@ -75,6 +75,36 @@ namespace Opm {
     }
 
 
+    void SimulationDataContainer::setCellDataComponent( const std::string& key , size_t component , const std::vector<int>& cells , const std::vector<double>& values) {
+        auto& data = getCellData( key );
+        if (component >= m_num_phases)
```

Comment:
This logic should be placed together with the check on line 92 since they depend on the same (temporary) restriction.